llm/providers/agent.py

```
@@ -0,0 +1,302 @@
+            env["HOME"] = home
+        env.pop("ANTHROPIC_API_KEY", None)
+        return env
+
+    def _parse_result(self, out, model_id, tier, usage):
+        """Extract data['result'] from the claude -p JSON envelope + record usage."""
+        try:
+            data = json.loads(out)
+        except json.JSONDecodeError as e:
+            raise RuntimeError(
+                f"claude_agent: failed to parse JSON envelope: {e}\nOutput: {out[-500:]}"
+            )
```

Ruff (0.15.17)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)